elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLMetricProjection.java

```diff
@@ -87,7 +87,8 @@ public SQLMetricProjection(Metric metric,
     @Override
     public String toSQL(Queryable source, SQLReferenceTable lookupTable) {
         if (expression.matches(".*\\{\\{.*\\}\\}.*")) {
-            return lookupTable.getResolvedReference(source, getName());
+            //TODO: Add query context to Table Context before calling get
```

Comment:
What does this mean?

Reply:
Changing method to `public String toSQL(Queryable source, SQLReferenceTable lookupTable, Map<String, Object> queryContext) {` and then use user provided queryContext for resolving definitions.